src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -154,13 +154,6 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
             // emitting it.
             dependencies.Add(new DependencyListEntry(_optionalFieldsNode, "Optional fields"));
 
-            if (factory.TypeSystemContext.HasLazyStaticConstructor(_type) && !_type.IsCanonicalSubtype(CanonicalFormKind.Any))
```

Comment:
We no longer need to root the class constructor context for all EETypes - only the constructed ones (assuming the invariant will be that only constructed EETypes are visible to user code from now on).